hnn_core/viz.py

```diff
@@ -516,3 +518,59 @@ def plot_psd(dpl, *, fmin=0, fmax=None, tmin=None, tmax=None, layer='agg',
 
     plt_show(show)
     return ax.get_figure()
+
+
+def _plot_cell_morphology(ax, cell_type):
+    if cell_type == 'L2Pyr':
+        sec_pts, _, _, _, _ = _secs_L2Pyr()
+    elif cell_type == 'L5Pyr':
+        sec_pts, _, _, _, _ = _secs_L5Pyr()
+    else:
+        raise ValueError('Unrecognized cell type to plot')
+
+    for sec_pt in sec_pts.values():
+        xs = [pt[0] for pt in sec_pt]
+        ys = [pt[1] for pt in sec_pt]
+        zs = [pt[2] for pt in sec_pt]
+        ax.plot(xs, ys, zs, 'b.-')
+
+
+def plot_cell_morphology(axes=None, cell_types=None, show=True):
```

Comment:
Hmm, I worry that this falls outside the scope of hnn-core. How do we plan on updating this function when we add new classes of neurons with varying amounts of morphological detail?

I think our best plan of action would be to convert the cell morphology code to NeuroML and demo its use in an example using [pyneuroml](https://docs.neuroml.org/Userdocs/VisualisingNeuroMLModels.html).

Reply:
you don't have to update this function when you add a new class of neuron. It's pretty general! See how I [plotted a neuron from an SWC file](https://github.com/jonescompneurolab/hnn-core/issues/165#issue-689520960) here. The brunt of the work will be to create an API which you'll have to do in any case if you decide to use another software.

Reply:
I was imagining neural models that compartment diameter detail (i.e., complex conical sections) as well. At the heart of it, I guess I'm having a hard time with the use-case for this. Since HNN isn't intended to help users design and construct neural models, but rather to interpret M/EEG signals emerging from preexisting models, I don't see the benefit in maintaining our own morphology visualization code. There are a bunch of other packages that specialize in model visualization that we could integrate or reference. Just to be clear, I love having the ability to plot the neuron morphology and can't deny that the simplicity and flexibility of this method is tantalizing. NBD though if others disagree.

Reply:
I think visualising the model is valuable, though I too will look forward to a more detailed geometric representation (reason to keep `Axes3D` even though only 2D projection shown). You'll /always/ need to describe the model to a new user/collaborator, so let's just make an example of it!

Reply:
To me, it's valuable just to know what the actual construction of the neuron looks like ... for educational purposes. What we have on HNN website are all schematics, they are not drawn to scale. I'm also using this as an excuse to clean out/separate code that is related to the geometry. 

Staying native to matplotlib also has its benefits for maintenance. Simple and universal API ... even people from Matlab world can follow. But hey, let me see if adding the diameter detail adds too much complexity. I think that should just be a couple of lines ...

Reply:
Fair enough. Morphology visualization even just as a sanity check is worth it, especially if we keep the plotting functions simple.